hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -60,24 +92,38 @@ func (defaulter *nodePoolDefaulter) Default(ctx context.Context, obj runtime.Obj
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a NodePool but got a %T", obj))
 	}
 
-	if np.Spec.Release.Image != "" {
-		return nil
-	} else if np.Spec.ClusterName == "" {
-		return fmt.Errorf("nodePool.Spec.ClusterName is a required field")
-	}
+	if np.Spec.Release.Image == "" {
+		if np.Spec.ClusterName == "" {
+			return fmt.Errorf("nodePool.Spec.ClusterName is a required field")
```

Comment:
Do you want to move it to the validation WH?

Reply:
mutation occurs before the validating webhook. This field is actually required by the CRD, but i think it's technically possible to get to this point before any validation takes place.